[Merged by Bors] - fix readme (#6068)
spacemeshos/go-spacemesh: README.md

### Setting Up Local Dev Environment

> Building is supported on OS X, Linux, FreeBSD, and Windows, which use GLIBC 2.34+ version

Reviewer: ```suggestion
Building is supported on OS X, Linux, FreeBSD, and Windows

In the case of Linux, GLIBC 2.34+ is required.
```

### Setting Up Local Dev Environment

Building is supported on OS X, Linux, FreeBSD, and Windows

> In the case of Linux, GLIBC 2.34+ is required.

Reviewer 2: More specifically we support

- Ubuntu 22.04 and newer (but any distro with GLIBC 2.34+ should work)
- MacOS 13 (Intel) and MacOS 14 (Arm) and newer
- Windows 10 and newer
- FreeBSD is not officially supported

make install
```

> Ensure, that environment is set correctly:

Reviewer: ```suggestion
Make sure the environment is set up correctly
```

```bash
make go-env-test
```

**CGO_CFLAGS** must be set to "-I<absolute_path_to_repo>/go-spacemesh/build/ -DSQLITE_ENABLE_DBSTAT_VTAB=1"
**CGO_LDFLAGS** must be set to "-L<absolute_path_to_repo>/go-spacemesh/build/ -Wl,-rpath,$ORIGIN -Wl,-rpath,<absolute_path_to_repo>/go-spacemesh/build/"

> Ensure you have installed **OpenCL** library

Reviewer: ```suggestion
Make sure you have **OpenCL** library installed
```
but also we probably want to have version specified

Reviewer 3: Both "make sure" and "ensure" are correct.

To check if setup was configured successfully, try to run: 
```bash
make test
```

> You should not see compilation errors, but running test will take some time. 

Reviewer: ```suggestion
There shouldn't be any build errors, but please note that running the tests will take some time.
```
